pkg/clusteragent/api/v1/types.go

```diff
@@ -50,13 +50,13 @@ func (m NamespacesPodsStringsSet) DeepCopy(old *NamespacesPodsStringsSet) Namesp
 	if old == nil {
 		return m
 	}
-
-	for key1, val1 := range *old {
-		for key2, val2 := range val1 {
-			if _, ok := m[key1][key2]; !ok {
-				m[key1] = MapStringSet{}
-			}
-			m[key1][key2] = sets.NewString(val2.List()...)
+	// {"ns":{"pod":{"svc"}, "pod1": {"svc"}, "pod2": {"svc1"}}}
+	for nsKey, val1 := range *old {
+		if _, ok := m[nsKey]; !ok {
+			m[nsKey] = MapStringSet{}
+		}
+		for pod, svcs := range val1 {
```

Comment:
```suggestion
		for pod, svcs := range val1 {
            if _, ok := m[nsKey][pod]; !ok {
                m[nsKey][pod] = sets.NewString()
            }
            m[nsKey][pod] = m[nsKey][pod].Union(val2)
        }
```

Reply:
Ok, that means relying on the consistency of the delete though, otherwise we'll end up with deleted items in the map.